### DISCUSSION TITLE
Uptime Monitoring Now in Open Beta 🚀🚀🚀

### DISCUSSION THREAD

**Discussion by gaprl** [getsentry/sentry#78100] [Launch Feedback]
We’re excited to announce Sentry's Uptime Monitoring is now available in open beta, enabling you effortlessly track the availability and reliability of your web services. Once configured, it continuously monitors your configured URLs, providing instant alerts and detailed insights to help you identify and resolve downtime promptly.

## Set Up During Open Beta

Out of the box, Sentry [automatically sets up](https://docs.sentry.io/product/alerts/uptime-monitoring/automatic-detection/) Uptime Monitoring for the most frequently encountered hostname in your error data, ensuring continuous monitoring of your most critical services.

You can also create custom uptime monitoring alerts for specific URLs, with full control over request details like HTTP method, headers, and body parameters. [Learn more how to set up custom uptime alerts](https://docs.sentry.io/product/alerts/create-alerts/) or [click here to create an alert](https://sentry.io/orgredirect/alerts/new/uptime).

![uptime-create-form](https://github.com/user-attachments/assets/2bbcca10-fd41-410e-af8d-39d623098982)

## How Uptime Monitoring Works

Sentry’s Uptime Monitoring performs health checks on configured URLs every minute. If an issue is detected, you'll be notified immediately, such as when:

- The URL returns a status code other than 200 (success).
- The response takes longer than 10 seconds (timeout).
- A DNS resolution error occurs.

By leveraging [distributed tracing](https://docs.sentry.io/concepts/key-terms/tracing/distributed-tracing/), Sentry enables you to pinpoint any errors that occur during an uptime check, streamlining triage and speeding up root cause identification. [Learn more](https://docs.sentry.io/product/alerts/uptime-monitoring/).

![uptime-issue](https://github.com/user-attachments/assets/cec60333-b813-4837-b6f0-43e9aeb80199)

## 💬 We Want to Hear Your Feedback

If you have any thoughts or suggestions, please join the conversation below.👇

*In case you're facing any issues, please open a [new Github Issue](https://github.com/getsentry/sentry/issues/new/choose).*

**Comment by frank-nankivell** [comment]
Hi Team,

Really looking forward to using this feature on our platform - and ideally a set of Alerts we can the push to some external web-hooks. The feature appears to be removed from the UI. Aware this is in open beta so just checking on timelines for it returning?

Thanks!

**Reply by gaprl** [reply]
Hey @frank-nankivell, we had to temporarily disable the creation of new uptime alerts yesterday but we're back on. Feel free to [click here to create a new uptime alert](https://sentry.io/orgredirect/alerts/new/uptime).

**Comment by gavinjohnson-crowdcube** [comment]
Are there any limitations or requisites to enable this?  If I log into Sentry and navigate to https://[company].sentry.io/alerts/new/uptime , enter all the details and when I then hit the Create Alert button I get a 404 back from the POST to https://us.sentry.io/api/0/projects/[company]/[whatever project I'm in]/uptime/

**Reply by gaprl** [reply]
Hi @gavinjohnson-crowdcube, we had to temporarily disable the creation of new uptime alerts yesterday but we're back on. Feel free to [click here to create a new uptime alert](https://sentry.io/orgredirect/alerts/new/uptime).

**Reply by gavinjohnson-crowdcube** [reply]
Yep, all working now!

**Comment by Anton-Aracor** [comment]
We have tow envs. But this alert is created automatically and sent notification to our PROD Slack channel, while pointing that DEV was downtime. In the configuration of the automatically created alert I see no option to change environment. And if I try to change url from http to https, for example, it din't save the change.

I need to have two separate alerts - one for PROD and one for DEV env. Didn't find how to do this configuration.

**Reply by gaprl** [reply]
Hey @Anton-Constructor, we had to temporarily disable the creation of new uptime alerts yesterday but we're back on. That's probably why you weren't able to save it. Can you please give it another try and let me know if you still face any issues?

**Comment by markmono** [comment]
Is there any simple way to exclude these events from the Discover events, as they spam it during my quiet times. 

I do this for AWS ELB health checks by having a custom traces sampler in the python sentry_sdk checking for user_agent ELB-HealthChecker/2.0 and ignoring which I could also add SentryUptimeBot/1.0, but is there a better way to do this using a built in x-header which is always ignored by the sdk?

**Reply by gaprl** [reply]
Hey @markmono, yes using the `User-Agent` to filter uptime transactions in your SDK's tracer is the best way to go. Our SDKs do not have a built-in header that automatically ignores recording transactions.

**Comment by th3hamm0r** [comment]
For smaller plans, this does not seem to be an option, as these uptime requests seem to be counted 1:1 towards the transaction/performance-unit quota.
Are there any plans to change this, or to allow a higher interval (e.g. once per hour)?

**Reply by gaprl** [reply]
Hey @th3hamm0r, yes we'll be adding support for changing the interval.

Regarding the requests being counted towards your transaction quota, we recommend [filtering](https://docs.sentry.io/platforms/javascript/configuration/filtering/) on the client-side to not send any events from our uptime bot. You can use the request`User-Agent` to check if it's a Sentry bot. Do note that by not sending these transactions to Sentry, you won't be able to see its traces when an uptime issue is created.

**Comment by smo-qarma** [comment]
Looks quite interesting. As soon as you have any indicators for pricing, please let us know!

Also, do you plan to have uptime checks performed from different data centers around the world? We would be particularly interested in Asia and China...

**Reply by gaprl** [reply]
Hi @smo-qarma, we are considering offering the ability to run uptime checks in different regions around the world. If you can, I'd love to learn more about your use case. How would such feature help you and your team? What are some of the metrics and functionality you'd expect?

**Comment by mridang** [comment]
The feature is neat. Is there a way to configure the interval? It currently polls every second. I couldn't find such an option.

**Reply by gaprl** [reply]
Hey @mridang, we are shipping the ability to update the interval very soon. I'll post an update here as soon as it is live.

Now, you shouldn't be seeing checks running every second (they should be every minute). If that's the case, could you share with us the URL of your monitor? I can take a look into it. Feel free to email us at uptime-feedback@sentry.io.

**Reply by gaprl** [reply]
Hey @mridang, just updating on my previous comment, the ability to update the interval is live and can be edited directly in your alert configuration.

**Reply by mridang** [reply]
Works like a charm

**Comment by smo-qarma** [comment]
Hi, thanks for getting back. We have a heavy presence in Asia, particularly China. Our goal is to make sure our availability in those regions is stable. So measuring uptime and also average response time. Today we use a simple service to do this which will perform a few retries if a location experiences an outage, and then we are alerted if it is a consistent outage. This is very basic. China is a story on its own in terms of access due to the Great Firewall, it is not straightforward to set up anything inside China. I will be happy to jump on a call if you need more info. We are very happy with Sentry and would happily help with input on moving more responsibility onto Sentry.
________________________________

**Comment by sandstrom** [comment]
I love Sentry, but the design of this feature seems really weird.

The main building block is an alert?

Generally how other uptime monitoring solutions work, is that you create the monitor with all it's details (URL, response checks, IPv6/IPv4, regions, etc) and you can then (optionally) tie alerts to those monitors.

But all uptime monitors doesn't have alerts tied to them, and also the criteria may differ. So e.g. you might want to run the monitor from 5 regions but only alert if its down from more than 3 at the same time.

It's also important with data insight, i.e. you need to be able to see for at least the past 24 hours or so what responses you are getting back.

I don't know if I've misunderstood how Sentry has designed it, but to me it looks like they haven't even looked at any of the myriad of existing tools for this (BetterUptime, Cronitor, Pingdom, Uptimerobot, etc).

**Reply by gaprl** [reply]
Hey @sandstrom, we really appreciate your feedback! We're taking a look how we can improve our UX, but rest assured some of the things you've mentioned are already on our roadmap, such as enhanced configurability for your monitors and history of checks.

**Reply by sandstrom** [reply]
Sounds good! I'll keep my eyes on the changelog. 

Overall very happy with Sentry!

**Comment by rtshilston** [comment]
Can you provide instructions for how to see the performance over time of a given monitor?  eg if I've set it to check api.mycompany.com/v1/health, how can I see the average response time / uptime / geographic split?  I'd love to migrate from Pingdom to Sentry for this.

**Comment by Digital39999** [comment]
(public) Status pages soon? 👀

**Reply by black-snow** [reply]
@gaprl this is what's holding us back from switching over to Sentry for uptime monitoring. Public status pages would be awesome - uptime checks look great.

**Comment by dcmidhun** [comment]
Is the uptime monitor feature available on self-hosted sentry, if not 

1. is there any other way we can do this in sentry self-hosted ?
2. When can we expect this feature to be released for self-hosted ?

**Comment by la5942** [comment]
@gaprl Any plans in the pipeline for scheduled maintenance windows?

**Reply by gaprl** [reply]
Yes, we have an issue tracking this feature request: https://github.com/getsentry/sentry/issues/84981 feel free to 👍 it and leave any comments.

**Comment by mdefeche** [comment]
I would love to have this metric in the dashboard and be able to use it to report the availability of our services to our clients (ie. be able to get the availability for a specific period of time)

Ability to add header is handy. I could adapt my WAF rules to allow traffic based on a custom header and value

**Comment by iloveitaly** [comment]
Can I configure alerting via the terraform provider? I'm not seeing docs for this.

**Comment by dtkav** [comment]
This feature is really annoying for servers that are set up to auto-stop when there is no network activity [0].
When i was setting up sentry I didn't expect it to make requests to my service at all.

Can you make this opt-in instead of opt-out? Or least ask for blanket user consent before sending network requests against their services?

[0] https://fly.io/docs/launch/autostop-autostart/

**Comment by NFreybier** [comment]
We like the monitoring function. But there is no indication of how much this feature costs, except on the announcement page. An entry on the subscription page would be helpful.
Additionally, we would appreciate it if the current monitoring result could also be seen on the project details page. After all, you also connect the monitoring to a project

**Comment by fnagel** [comment]
Can we have ipv4 / ipv6 checks? Just had an issue with ipv6 not working and all my monitoring said its fine, as only IPv4 was tested...
